app/services/author/repository/create_with_book.rb

```diff
@@ -20,20 +20,25 @@ def call
   def create_author_with_book
     ActiveRecord::Base.transaction do
       author = Author.create!(params)
-      author.books.create!(book_params.merge(publisher: author))
+      author.books.create!(book_params(publisher: author))
     end
   rescue ActiveRecord::RecordInvalid
     answer.fail("Can't update author")
     false
   end
 
-  def book_params
-    @book_params || dummy_book_params
+  def book_params(publisher:)
+    if @book_params.is_a? Array
+      return @book_params.map { |params| params.merge(publisher: publisher) }
+    end
+
+    params = @book_params || dummy_book_params
+    params.merge(publisher: publisher)
```

Comment:
This code is a little confusing, since you're handling assigning the `publisher` in two places. It would be clearer written like this:

```suggestion
    Array[@book_params || dummy_book_params].map { |params| params.merge(publisher: publisher) }
```

The `Array[]` syntax will cast the input to an array, or leave it alone if it's already an array.